tests/integration/targets/azure_rm_devtestlab/tasks/main.yml

```diff
@@ -144,6 +145,99 @@
     that:
       - output.changed
 
+- name: Create instance of DTL Virtual Machine
+  azure_rm_devtestlabvirtualmachine:
+    resource_group: "{{ resource_group }}"
+    lab_name: "{{ lab_name }}"
+    name: "{{ vm_name }}"
+    notes: Virtual machine notes, just something....
+    os_type: linux
+    size: Standard_A2_v2
+    user_name: dtladmin
+    password: ZSuppas$$21!
+    lab_subnet_name: "{{ vn_name }}Subnet"
+    lab_virtual_network_name: "{{ vn_name }}"
+    disallow_public_ip_address: no
+    image:
+      offer: UbuntuServer
+      publisher: Canonical
+      sku: 16.04-LTS
+      os_type: Linux
+      version: latest
+    artifacts:
+      - source_name: "{{ artifacts_name }}"
+        source_path: "/Artifacts/linux-install-mongodb"
+    allow_claim: no
+    expiration_date: "2019-02-22T01:49:12.117974Z"
+  register: vm_output
+
+- name: Assert that change was registered
+  assert:
+    that:
+      - output.changed
+
+- name: Update instance of DTL Virtual Machine with same parameters
+  azure_rm_devtestlabvirtualmachine:
+    resource_group: "{{ resource_group }}"
+    lab_name: "{{ lab_name }}"
+    name: "{{ vm_name }}"
+    notes: Virtual machine notes, just something....
+    os_type: linux
+    size: Standard_A2_v2
+    user_name: dtladmin
+    password: ZSuppas$$21!
+    lab_subnet_name: "{{ vn_name }}Subnet"
+    lab_virtual_network_name: "{{ vn_name }}"
+    disallow_public_ip_address: no
+    image:
+      offer: UbuntuServer
+      publisher: Canonical
+      sku: 16.04-LTS
+      os_type: Linux
+      version: latest
+    artifacts:
+      - source_name: "{{ artifacts_name }}"
+        source_path: "/Artifacts/linux-install-mongodb"
+    allow_claim: no
+    expiration_date: "2019-02-22T01:49:12.117974Z"
+  register: output
+
+- name: Assert that nothing has changed
+  assert:
+    that:
+      - output.changed == false
+
+- name: Update instance of DTL Virtual Machine - change notes
+  azure_rm_devtestlabvirtualmachine:
+    resource_group: "{{ resource_group }}"
+    lab_name: "{{ lab_name }}"
+    name: "{{ vm_name }}"
+    notes: Virtual machine notes, just something.... more text
+    os_type: linux
+    size: Standard_A2_v2
+    user_name: dtladmin
+    password: ZSuppas$$21!
+    lab_subnet_name: "{{ vn_name }}Subnet"
+    lab_virtual_network_name: "{{ vn_name }}"
+    disallow_public_ip_address: no
+    image:
+      offer: UbuntuServer
+      publisher: Canonical
+      sku: 16.04-LTS
+      os_type: Linux
+      version: latest
+    artifacts:
+      - source_name: "{{ artifacts_name }}"
+        source_path: "/Artifacts/linux-install-mongodb"
+    allow_claim: no
+    expiration_date: "2019-02-22T01:49:12.117974Z"
```

Comment:
is it ok to set hardcode expiration time, what if test after that date? will it fail?

Reply:
changed to 2029 just in case. we should be safe ;-)